src/resources/pulse_counter_esp32.cc

```diff
@@ -268,8 +271,7 @@ PRIMITIVE(new_unit) {
 
 PRIMITIVE(close_unit) {
   ARGS(PcntUnitResource, unit)
-  unit->tear_down();
-  pcnt_unit_ids.put(unit->unit_id());
+  unit->resource_group()->unregister_resource(unit);
```

Comment:
It feels odd that we're overwriting on_unregister_resource in the group instead of just letting the tear_down functionality be the destructor.

Reply:
done.